daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2258,6 +2281,48 @@ func (devices *DeviceSet) exportDeviceMetadata(hash string) (*deviceMetadata, er
 	return metadata, nil
 }
 
+// Return true only if kernel supports xfs and mkfs.xfs is available
+func supportsXFS() bool {
+        exec.Command("modprobe", "xfs").Run()
+
+        f, err := os.Open("/proc/filesystems")
+        if err != nil {
+                return false
+        }
+        defer f.Close()
+
+        s := bufio.NewScanner(f)
+        for s.Scan() {
+                if strings.Contains(s.Text(), "xfs") {
+			if _, err := exec.LookPath("mkfs.xfs"); err == nil {
```

Comment:
we're already under the condition where we found `xfs`, so if `mkfs.xfs` does not exist, it won't exist later either, so we can simply `return err == nil` which would return `false` if `xfs` is present in `/proc/filesystems` but no `mkfs.xfs` is in PATH.